src/components/Accounts.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import type { KeyringPair } from '@polkadot/keyring/types';
+import React from 'react';
+import { Container, Dropdown } from 'semantic-ui-react';
+import styled from 'styled-components';
+
+import AccountActions from '../actions/accountActions';
+import { useAccountContext, useUpdateAccountContext } from '../contexts/AccountContextProvider';
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+import useAccounts from '../hooks/useAccounts';
+import useDerivedAccount from '../hooks/useDerivedAccount';
+
+interface Props {
+  className?: string;
+}
+
+const formatOptions = (accounts: Array<KeyringPair>) =>
+  accounts.map(({ meta, address }) => ({
+    icon: 'user',
+    key: address,
+    text: (meta.name as string).toLocaleUpperCase(),
+    value: address
+  }));
+
+const Accounts = ({ className }: Props) => {
+  const accounts = useAccounts();
+  const derivedAccount = useDerivedAccount();
+  const { dispatchAccount } = useUpdateAccountContext();
+  const { sourceChain, targetChain } = useSourceTarget();
+  const { account: currentAccount } = useAccountContext();
+
+  const value = currentAccount?.address || '';
+
+  const onChange = (evt: any, { value }: any) => {
+    const account = accounts.find(({ address }) => address === value);
+    if (account) {
+      dispatchAccount({ payload: { account }, type: AccountActions.SET_ACCOUNT });
```

Comment:
Is it really necessary to complicate this so much (i.e. payload & action)? Why not simply have `setAccount` function coming from `useAccounts`?

Reply:
The reason why i'm using a global variable for account is because the account object is ( and will be ) used in a lot of components across the UI. If i just rely on a local state for that , then i would have to pass the selected account in a props chain hell.

Reply:
You're missing the point here. I'm fine with using context for storing this, don't want this do be pushed through the components.
My point was that I think that the reducer API (and action construction) is pretty low-level API and shouldn't be used in the high-level component (`Accounts.tsx`). What I proposed in some other comment is to create account-related hook and only expose `setCurrentAccount` function - it's fine to use the reducer under the hood, but the point is to expose simple API for high-level components.